File: src/algorithms/strings/palindrome-check/solution3.go
package main

// O(n) time | O(n) space
func IsPalindrome(str string) bool {
	return helper(str, 0)
}

func helper(str string, i int) bool {
	j := len(str) - 1 - i
	if i >= j {
		return true
	}
	if str[i] != str[j] {
		return false
	}
	return helper(str, i+1)
}

File: src/algorithms/strings/palindrome-check/solution3.js
// O(n) time | O(n) space
function isPalindrome(string, i = 0) {
	const j = string.length - 1 - i;
	return i >= j ? true : string[i] === string[j] && isPalindrome(string,i + 1);
};

exports.isPalindrome = isPalindrome;